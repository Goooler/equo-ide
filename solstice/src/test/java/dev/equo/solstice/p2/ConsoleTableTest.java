/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.solstice.p2;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class ConsoleTableTest {
	private P2Query query() throws Exception {
		var session = new P2Session();
		try (var client = new P2Client()) {
			session.populateFrom(client, "https://download.eclipse.org/eclipse/updates/4.25/");
		}
		var query = session.query();
		query.addAllUnits();
		return query;
	}

	@Test
	public void detail(Expect expect) throws Exception {
		var query = query();
		expect.toMatchSnapshot(
				ConsoleTable.detail(
						query.findAllAvailableUnitsById(
								"org.eclipse.ecf.filetransfer.httpclient5.feature.feature.jar"),
						ConsoleTable.Format.ASCII));
	}
}
