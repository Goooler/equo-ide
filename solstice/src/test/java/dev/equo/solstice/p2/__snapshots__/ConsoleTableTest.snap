dev.equo.solstice.p2.ConsoleTableTest.detail=[
+-----------------------------------------+--------------------------------------------------------------------------+
| key                                     | value                                                                    |
+-----------------------------------------+--------------------------------------------------------------------------+
| id                                      | org.eclipse.ecf.filetransfer.httpclient5.feature.feature.jar             |
| version                                 | 1.1.600.v20220215-0126                                                   |
| maven coordinate                        | org.eclipse.ecf:org.eclipse.ecf.filetransfer.httpclient5.feature:1.1.600 |
| maven repo                              | mavenCentral?                                                            |
| filter                                  | (org.eclipse.update.install.features=true)                               |
| prop maven-artifactId                   | org.eclipse.ecf.filetransfer.httpclient5.feature                         |
| prop maven-groupId                      | org.eclipse.ecf                                                          |
| prop maven-version                      | 1.1.600-SNAPSHOT                                                         |
| prop org.eclipse.equinox.p2.description | This feature provides the Apache   HttpComponents/HttpClient 5 based     |
|                                         |   FileTransfer   provider used by the Eclipse platform to support   P2   |
|                                         |   filetransfer.                                                          |
| prop org.eclipse.equinox.p2.name        | ECF Apache Httpclient 5 FileTransfer Provider                            |
+-----------------------------------------+--------------------------------------------------------------------------+

]